gcp/src/main/java/org/apache/iceberg/gcp/biglake/BigLakeCatalog.java

```diff
@@ -0,0 +1,404 @@
+    client.updateDatabaseParameters(dbName, optionsBuilder.getParametersMap());
+    return true;
+  }
+
+  @Override
+  public boolean removeProperties(Namespace namespace, Set<String> props) {
+    DatabaseName dbName = databaseName(namespace);
+    if (dbName == null) {
+      throwInvalidDbNamespaceError(namespace);
+    }
+
+    HiveDatabaseOptions.Builder optionsBuilder =
+        loadDatabase(namespace).toBuilder().getHiveOptionsBuilder();
+    props.forEach(optionsBuilder::removeParameters);
+    client.updateDatabaseParameters(dbName, optionsBuilder.getParametersMap());
+    return true;
+  }
+
+  @Override
+  public Map<String, String> loadNamespaceMetadata(Namespace namespace) {
+    if (namespace.isEmpty()) {
+      // Calls getCatalog to check existence. BLMS catalog has no metadata today.
+      client.getCatalog(catalogName);
+      return ImmutableMap.of();
+    } else if (namespace.levels().length == 1) {
+      return loadDatabase(namespace).getHiveOptions().getParametersMap();
+    } else {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+  }
+
+  @Override
+  public String name() {
+    return name;
+  }
+
+  @Override
+  protected Map<String, String> properties() {
+    return properties == null ? ImmutableMap.of() : properties;
+  }
+
+  @Override
+  public void setConf(Object conf) {
+    this.conf = conf;
+  }
+
+  @Override
+  public void close() throws IOException {
+    if (closeableGroup != null) {
+      closeableGroup.close();
+    }
+  }
+
+  @Override
+  protected boolean isValidIdentifier(TableIdentifier tableIdentifier) {
+    return tableIdentifier.namespace().levels().length == 1;
+  }
+
+  private String databaseLocation(String dbId) {
+    return String.format("%s/%s.db", warehouseLocation, dbId);
+  }
+
+  private static TableIdentifier tableIdentifier(Table table) {
+    TableName tableName = TableName.parse(table.getName());
+    return TableIdentifier.of(Namespace.of(tableName.getDatabase()), tableName.getTable());
+  }
+
+  private static Namespace namespace(Database db) {
+    return Namespace.of(DatabaseName.parse(db.getName()).getDatabase());
+  }
+
+  private TableName tableName(String dbId, String tableId) {
+    return TableName.of(projectId, region, catalogId, dbId, tableId);
+  }
+
+  private String databaseId(Namespace namespace) {
+    return namespace.levels().length == 1 ? namespace.level(0) : null;
+  }
+
+  private DatabaseName databaseName(Namespace namespace) {
+    String db = databaseId(namespace);
+    return db == null ? null : DatabaseName.of(projectId, region, catalogId, db);
+  }
+
+  private Database loadDatabase(Namespace namespace) {
+    return client.getDatabase(databaseName(namespace));
+  }
+
+  private void throwInvalidDbNamespaceError(Namespace namespace) {
+    throw new NoSuchNamespaceException(
+        "Invalid BigLake database namespace: %s", namespace.isEmpty() ? "empty" : namespace);
```

Comment:
No need for "BigLake database" here. Just "Invalid namespace" is enough.

Reply:
Done.